src/decisionengine/framework/engine/DecisionEngine.py

```diff
@@ -130,12 +130,14 @@ def _dispatch(self, method, params):
         return func(*params)
 
     def block_while(self, state, timeout=None):
+        self.logger.debug(f"Waiting for {state} or timeout={timeout} on channel_workers.")
         with self.channel_workers.unguarded_access() as workers:
             if not workers:
                 self.logger.info("No active channels to wait on.")
                 return "No active channels."
             for tm in workers.values():
                 if tm.is_alive():
+                    self.logger.debug(f"checking for state={state} in {tm.name}.")
```

Comment:
Are you wanting the name of the process (`tm.name`) or the name of the task manager (`tm.task_manager.name`)?

Reply:
Which one do you think would be most useful for isolating a worker that got stuck (never reached `state`)?  The process name felt easy, but digging deeper into the task_manager might be cleaner.....

Reply:
I suggest using `tm.task_manager.name` instead--that is a more accurate reflection of whose state we're waiting on.  Makes me wonder if we should add the debug statement to the `Worker` (i.e. one level down) instead of the `DecisionEngine`.

Reply:
Pat do you mind using something like:

```suggestion
                    self.logger.debug(f"Waiting for {tm.task_manager.name} to exit {state} state.")
```

Reply:
Changed merged in.